packages/docs/blog/2025-04-22-multi-tenant-mso.md

```diff
@@ -0,0 +1,293 @@
+---
+slug: multi-tenant-mso
+title: Multi-Tenant MSO with Medplum
+authors: finnbergquist
+tags: [mso, tenant, access-control]
+---
+
+# A step by step guide to building a Multi-Tenant Managed Service Organization with Medplum
+
+A common pattern in the Medplum community is to build a platform that **serves multiple clinics** in the form of a Managed Service Organization (MSO).
+
+In this post, we'll explore how to build a robust MSO platform using Medplum, focusing on [multi-tenancy](#tenant-management-with-organizations), [access control](#access-control), [patient consent management](#additional-access-control-patient-consent), and [project-scoped users](#user-management-strategy).
+
+If you haven't checked out the **MSO Demo App** yet, you can view the code [here](https://github.com/medplum/medplum-mso-demo) or watch our demo video below.
+
+<div className="responsive-iframe-wrapper">
+  <iframe src="https://www.youtube.com/embed/FJIZTI9_fBc?start=0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+</div>
+
+
+## Understanding the MSO Architectural Needs
+
+
+At its core, an MSO usually needs to handle the following requirements:
+- Multiple clinics (tenants) sharing the same platform
+- [Enrollment workflows](https://github.com/medplum/medplum-mso-demo/blob/main/src/operations/enrollment.ts) that allow Practitioners and Patients to be added and removed from one or more clinics
```

Comment:
Link to here https://www.medplum.com/docs/administration/provider-directory and add comment related